### PR TITLE
Favor Calculator Implementation

### PR DESCRIPTION
**Pull request by nick-mynatt**
More verbose description in each individual commit

### CODE REVIEW

File: ndn_hydra/repo/group_messages/heartbeat.py [hunk 1]
```diff
@@ -13,14 +13,17 @@
...
 class HeartbeatMessageTypes:
     NODE_NAME = 84
-    FAVOR = 86
+    LATITUDE = 85
+    LONGITUDE = 86
 class HeartbeatMessageTlv(TlvModel):
     node_name = BytesField(HeartbeatMessageTypes.NODE_NAME)
-    favor = BytesField(HeartbeatMessageTypes.FAVOR)
+    latitude = BytesField(HeartbeatMessageTypes.LATITUDE)
+    longitude = BytesField(HeartbeatMessageTypes.LONGITUDE)
```
Comment (justincpresley) [inline ndn_hydra/repo/group_messages/heartbeat.py:26]: First, we need a structure called FavorParamaters that includes lat/long and other information.
We can then include that struct in ALL messages via the ModelType tlv. In addition, we need to update the global view every message.

Comment (nick-mynatt) [reply ndn_hydra/repo/group_messages/heartbeat.py]: Hi, please check out my most recent commit and let me know what you think. In regards to updating the global view, do you mean that just for the heartbeat messages? Or ALL messages? If you mean just the heartbeat, I believe in the apply function in the Heartbeat message it updates the global view with the new calculated favor.

Screenshot of 4 nodes running with different coordinates in client
The node table in global view of node1

**Comment by nick-mynatt** [comment]
Two commits since my last update:
1. Introduced FavorParameters class
2. Updated that class with better parameters for calculating favor

Here is an example of two nodes calculating the other's favor
Screenshot from 2022-09-28 18-09-58

**Comment by justincpresley** [comment]
Looks good. Could you run through the examples? You can just state when its good to merge.

**Comment by nick-mynatt** [comment]
Ran through examples with no issues. Should be good to merge!

Screenshot from 2022-09-30 23-14-56

**Pull request merged**